src/main/java/hw11and12/FileManagerNIO.java

```diff
@@ -0,0 +1,93 @@
+package hw11and12;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FileManagerNIO {
+    private static final Logger logger = Logger.getLogger(FileManagerIO.class.getName());
+    private Path currentDirectory = Paths.get("").toAbsolutePath();
+
+    public void run() {
+        try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+            currentDirectory = Path.of(System.getProperty("user.dir"));
+            while (true) {
+                logger.log(Level.INFO, "{0}>", currentDirectory);
+                String command = reader.readLine();
+                String[] tokens = command.split("\\s+");
+                switch (tokens[0]) {
+                    case "cd" -> changeDirectory(tokens);
+                    case "cp" -> copyFile(tokens);
+                    case "ls" -> listDirectory();
+                    case "pwd" -> printWorkingDirectory();
+                    case "exit" -> System.exit(0);
+                    default -> logger.log(Level.WARNING, "Unknown command: {0}", command);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error of input reading ", e);
+        }
+    }
+
+    private void changeDirectory(String[] tokens) {
+        if (tokens.length < 2) {
+            logger.log(Level.INFO, "Usage: cd <directory>");
+            return;
+        }
+        String path = tokens[1];
+
+        if (path.equals("..")) {
+            currentDirectory = currentDirectory.getParent();
+        } else if (path.startsWith("/")) {
+            currentDirectory = Paths.get(path);
+        } else {
+            currentDirectory = currentDirectory.resolve(path);
+        }
+        if (!Files.exists(currentDirectory) || !Files.isDirectory(currentDirectory)) {
+            logger.log(Level.INFO, "Directory does not exist: {0}", currentDirectory);
+            currentDirectory = currentDirectory.getParent();
+        }
```

Comment:
You should first check if the directory exists and if yes, then allow it to run your logic, otherwise you should return from the method. You did the same in the first if by checking `tokens.length < 2`